src/main/java/com/datasonnet/header/Header.java

```diff
@@ -250,6 +251,7 @@ private static Header parseHeader20(String headerSection) throws HeaderParseExce
         List<MediaType> dataformat = new ArrayList<>(4);
 
         for (String line : headerSection.split("\\r?\\n")) {
+            line = line.trim();  // we never care about leading or trailing whitespace
```

Comment:
Would it be possible to replace this with `line = line.trim().split("\\/\\/",1);`?
This way, if theres trailing comments it would just be dropped and lines starting with comments would be dropped as well. I would only see an issue if theres some header value that uses `//`

Then wrap the try block with an `if !line.isEmpty()` so we dont have an empty else if

Reply:
Unfortunately there's a good possibility of double slashes being used as a special character in headers at some point, since some data formats do have comments and we might well allow people to configure what comments are recognized (there're a lot of custom semi-structured data formats out there with inconsistent commenting practices), so I don't think we can do that without introducing potential future problems.

I tend to prefer the style with the empty else if, which means every line gets handled with code structured the same way.